addon/templates/components/bs3/bs-modal/header.hbs

```diff
@@ -1,8 +1,14 @@
-{{#if closeButton}}
-  {{component closeComponent onClick=(action onClose)}}
-{{/if}}
 {{#if hasBlock}}
```

Comment:
I wouldn't want to make this a breaking change.

What about this: 
* besides the special template keyword `hasBlock` there is also `has BlockParams`
* as previously we didn't yield any components, this usage can be covered with the `else` part of `{{#if hasBlockParams}}`
* the new `{{yield (hash ...)}}` moves into the if block of `{{#if hasBlockParams}}`

If I am not missing something, this way we could support the yielded components (without a close button by default), while still remaining backwards compatible.

What do you think?